tests/unit/plugins/become/test_ksu.py

```diff
@@ -10,30 +10,54 @@
 import re
 
 from ansible import context
-from ansible.playbook.play_context import PlayContext
-from ansible.plugins.loader import become_loader
+
+from .helper import call_become_plugin
 
 
 def test_ksu(mocker, parser, reset_cli_args):
     options = parser.parse_args([])
     context._init_global_context(options)
-    play_context = PlayContext()
 
     default_cmd = "/bin/foo"
     default_exe = "/bin/bash"
     ksu_exe = 'ksu'
     ksu_flags = ''
 
-    cmd = play_context.make_become_cmd(cmd=default_cmd, executable=default_exe)
-    assert cmd == default_cmd
+    success = 'BECOME-SUCCESS-.+?'
+
+    task = {
+        'become_user': 'foo',
+        'become_method': 'community.general.ksu',
+        'become_flags': ksu_flags,
+    }
+    var_options = {}
+    cmd = call_become_plugin(task, var_options, cmd=default_cmd, executable=default_exe)
+    print(cmd)
+    assert (re.match("""%s %s %s -e %s -c 'echo %s; %s'""" % (ksu_exe, task['become_user'], ksu_flags,
+                                                              default_exe, success, default_cmd), cmd) is not None)
+
+
+def test_ksu_varoptions(mocker, parser, reset_cli_args):
+    options = parser.parse_args([])
+    context._init_global_context(options)
+
+    default_cmd = "/bin/foo"
+    default_exe = "/bin/bash"
+    ksu_exe = 'ksu'
+    ksu_flags = ''
 
     success = 'BECOME-SUCCESS-.+?'
 
-    play_context.become = True
-    play_context.become_user = 'foo'
-    play_context.set_become_plugin(become_loader.get('ksu'))
-    play_context.become_method = 'ksu'
-    play_context.become_flags = ksu_flags
-    cmd = play_context.make_become_cmd(cmd=default_cmd, executable=default_exe)
-    assert (re.match("""%s %s %s -e %s -c 'echo %s; %s'""" % (ksu_exe, play_context.become_user, ksu_flags,
+    task = {
+        'become_user': 'foo',
+        'become_method': 'community.general.ksu',
+        'become_flags': ksu_flags,
+    }
+    var_options = {
+        'become_user': 'bar',
+        'become_flags': '',
+    }
+    cmd = call_become_plugin(task, var_options, cmd=default_cmd, executable=default_exe)
+    print(cmd)
+    assert (re.match("""%s %s %s -e %s -c 'echo %s; %s'""" % (ksu_exe, task['become_user'], '',
```

Comment:
this should match var_options['become_user'] and flags, not task, since they are overridden, i would have several  invocations and tests, each assert will have to match depending on inputs given